src/pages/profile.tsx

```diff
@@ -0,0 +1,64 @@
+import { useSession } from 'next-auth/react';
+import { useEffect, useState } from 'react';
+import { SignOutButton } from '../components/SignOutButton';
+import { Image } from '@chakra-ui/react';
+
+export default function Profile() {
+  const { data: session, status } = useSession();
+  const [numAbsences, setNumAbsences] = useState<number | null>(null);
+  const [usedAbsences, setUsedAbsences] = useState<number | null>(null);
+
+  useEffect(() => {
+    const fetchUserInfoByEmail = async () => {
+      if (!session || !session.user || !session.user.email) return;
+
+      const email = session.user.email;
+      const apiUrl = `/api/users/email/${email}?getAbsences=true`;
+
+      try {
+        const response = await fetch(apiUrl);
+        if (!response.ok) {
+          throw new Error(response.statusText);
+        }
+        const data = await response.json();
+        setNumAbsences(data['numOfAbsences']);
```

Comment:
This feels a little iffy. Maybe we can have an interface for the user data and include what params an absence has and then access the params with a period instead.

```ts
interface UserData {
  numOfAbsences: number;
  absences: Array<{ /* details of each absence */ }>;
}
```
```ts
const data: UserData = await response.json();
setNumAbsences(data.numOfAbsences);
setUsedAbsences(data.absences.length);
```

Reply:
Updated here! https://github.com/uwblueprint/sistema/pull/20/commits/26096395b7b447397f3bcd71e39d00d406ea40aa